### PR TITLE
Closes #77: Initialize 4337 directory

### LINKED ISSUE #77: [4337 Module] Setup 4337 module in modules repository
- Use hardhat with hardhat-deploy for counterfactual deployment
- Setup empty contracts and tests

### PR DESCRIPTION
**Pull request by rmeissner**
Closes #77

- Copy code from https://github.com/5afe/eip4337-diatomic/tree/gas_overhead_test#readme
- Add GitHub action
- Add Issue templates

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+---
+name: Bug report
+about: Bug report about the Safe smart contracts
+
+---
+
+## Prerequisites
+
+- First, many thanks for taking part in the community and helping us improve. We appreciate that a lot.
+- Support questions are better asked on our Discord: https://chat.gnosis-safe.io
```
Comment (akshay-ap) [inline .github/ISSUE_TEMPLATE/bug_report.md:10]: do we have a new link?

Comment (akshay-ap) [inline .github/ISSUE_TEMPLATE/bug_report.md:10]:
```suggestion
- Support questions are better asked on our Discord: https://chat.safe.global
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+- Please ensure the issue isn't already reported.
+
+*Please delete the above section and the instructions in the sections below before submitting*
+
+## Description
+
+Please describe considely the bug you have found, and what you expect instead.
```
Comment (akshay-ap) [inline .github/ISSUE_TEMPLATE/bug_report.md:17]:
```suggestion
Please describe consiely the bug you have found, and what you expect instead.
```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+---
+name: Feature request
+about: Suggest an idea or feature request for the Safe smart contracts project
+
+---
+
+## Prerequisites
+
+- First, many thanks for taking part in the community and helping us improve. We appreciate that a lot.
+- Support questions are better asked on our Discord: https://chat.gnosis-safe.io
```
Comment (akshay-ap) [inline .github/ISSUE_TEMPLATE/feature_request.md:10]:
```suggestion
- Support questions are better asked on our Discord: https://chat.safe.global
```

File: 4337/README.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
+# Safe Module/Fallback handler for EIP4337 Support
+
+The diagram below outlines the flow that is triggered when a user operation is submitted to via the entrypoint. Additionally the gas overhead compared to a native implementation is mentioned.
```
Comment (akshay-ap) [inline 4337/README.md:3]:
```suggestion
The diagram below outlines the flow that is triggered when a user operation is submitted to the entrypoint. Additionally the gas overhead compared to a native implementation is mentioned.
```

File: 4337/README.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+This command will upload the contract source to Etherescan
+
+```bash
+npx hardhat --network <network> etherscan-verify
+```
+
+## Documentation
+
+- [Safe developer portal](http://docs.gnosis-safe.io)
```
Comment (akshay-ap) [inline 4337/README.md:112]:
```suggestion
- [Safe developer portal](https://docs.safe.global)
```

**Review by akshay-ap — CHANGES_REQUESTED** [review]

**Review by akshay-ap — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 1 above)
Comment (rmeissner) [reply .github/ISSUE_TEMPLATE/bug_report.md:10]: Fixed in another PR

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 3 above)
Comment (rmeissner) [reply .github/ISSUE_TEMPLATE/feature_request.md:10]: Same

File: 4337/README.md (hunk 5 above)
Comment (rmeissner) [reply 4337/README.md:112]: Same

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 2 above)
Comment (rmeissner) [reply .github/ISSUE_TEMPLATE/bug_report.md:17]: Fixed in #80

**Pull request merged**